packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_view.dart

```diff
@@ -13,6 +18,173 @@ class SnapshotView extends StatelessWidget {
+  }
+}
+
+class _ClassNameColumn extends ColumnData<HeapStatsRecord> {
+  _ClassNameColumn()
+      : super(
+          'Class',
+          titleTooltip: 'Class name',
+          fixedWidthPx: scaleByFontFactor(100.0),
+          alignment: ColumnAlignment.left,
+        );
+
+  @override
+  String? getValue(HeapStatsRecord record) => record.className;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  String getTooltip(HeapStatsRecord record) => record.fullClassName;
+}
+
+class _InstanceColumn extends ColumnData<HeapStatsRecord> {
+  _InstanceColumn()
+      : super(
+          'Non GC-able\nInstances',
+          titleTooltip: 'Number of instances of the class, '
+              'that have retaining path from the root.',
+          fixedWidthPx: scaleByFontFactor(110.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(HeapStatsRecord record) => record.instanceCount;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+}
+
+class _ShallowSizeColumn extends ColumnData<HeapStatsRecord> {
+  _ShallowSizeColumn()
+      : super(
+          'Shallow\nSize',
+          titleTooltip: 'Total shallow Dart (not native) size of objects.',
+          fixedWidthPx: scaleByFontFactor(85.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(HeapStatsRecord record) => record.shallowSize;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+
+  @override
+  String getDisplayValue(HeapStatsRecord record) => prettyPrintBytes(
+        getValue(record),
+        includeUnit: true,
+        kbFractionDigits: 1,
+      )!;
+}
+
+class _RetainedSizeColumn extends ColumnData<HeapStatsRecord> {
+  _RetainedSizeColumn()
+      : super(
+          'Retained\nSize',
+          titleTooltip: 'Total size of objects plus objects they retain, '
+              'taking to account only the shortest retaining path for the referenced objects.',
+          fixedWidthPx: scaleByFontFactor(85.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(HeapStatsRecord record) => record.retainedSize;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+
+  @override
+  String getDisplayValue(HeapStatsRecord record) => prettyPrintBytes(
+        getValue(record),
+        includeUnit: true,
+        kbFractionDigits: 1,
+      )!;
+}
+
+class _StatsTable extends StatefulWidget {
+  const _StatsTable({
+    Key? key,
+    required this.data,
+    required this.sorting,
+  }) : super(key: key);
+  final List<HeapStatsRecord> data;
+  final ColumnSorting sorting;
+
+  @override
+  State<_StatsTable> createState() => _StatsTableState();
+}
+
+class _StatsTableState extends State<_StatsTable> {
+  late final List<ColumnData<HeapStatsRecord>> _columns;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
```

Comment:
this can all be done in initState. We only need to use didChangeDependencies instead of initState when we need to be able to access `context`